safe_cli/operators/hw_accounts/hw_account_manager.py

```diff
@@ -0,0 +1,140 @@
+from enum import Enum
+from typing import Dict, List, Optional, Set, Tuple
+
+from eth_typing import ChecksumAddress
+from prompt_toolkit import HTML, print_formatted_text
+
+from gnosis.eth.eip712 import eip712_encode
+from gnosis.safe import SafeTx
+
+from safe_cli.operators.hw_accounts.hw_account import HwAccount
+
+
+class HwWalletType(Enum):
+    TREZOR = "Trezor"
+    LEDGER = "Ledger"
+
+
+class HwAccountManager:
+    def __new__(cls):
+        if not hasattr(cls, "instance"):
+            cls.instance = super(HwAccountManager, cls).__new__(cls)
```

Comment:
Please don't use `__new__` hacks, we already had enough problems with that on other projects (somehow I decided that was a good idea some time ago).

Use an auxiliary external function to the class like `get_hw_account_manager()`, which is `cached` so we always return the same singleton. Also for mocking and testing is easier

Reply:
Ok, no issues with that, but use `__new__` is the more common way to do a singleton class in python 
https://www.geeksforgeeks.org/singleton-pattern-in-python-a-complete-guide/

Reply:
I feel the factory function is more clear and less hacky. Also remember in Python 3 there's no need to pass arguments to the `super()` function